### DISCUSSION TITLE
XAML syntax for binding to a certain ViewModel type in logical parent's DataContext

### DISCUSSION THREAD

**Discussion by MihaMarkic** [AvaloniaUI/Avalonia#8709] [Ideas]
Imagine a situation where you have parent and children, like the one below (pseudo AXAML), where ParentViewModel has RemoveCommand present that would remove the child.
While the syntax _{Binding #Parent.DataContext.RemoveCommand}_ should work, there are some problems with it. First, you have to name the parent control, then child view can be hosted in different parent views etc.

I was thinking that something like _$parent_ - but instead of searching for parent type, it'd be searching for parent's DataContext type - could be helpful.
XAML binding could be something like "_{Binding $parentContext[ParentViewModel].RemoveCommand}_".

```xaml
<ItemsControl x:Name="Parent" Items="{Binding Items}" DataType="MasterViewModel">
  <ItemsControl.DataTemplates>
    <DataTemplate DataType="ChildViewModel">
       ...
       <Button Command="{Binding #Parent.DataContext.RemoveCommand}">Remove</Button>
    </DataTemplate>
  </ItemsControl.DataTemplates>
</ItemsControl>
```

**Comment by jp2masa** [comment]
I used to want this feature too, but it's probably much more work than it's worth it, because:

1. As you show in your example, you can bind to a named element
2. This typically occurs on owner-owned relationships, so why not pass the owner viewmodel to the owned one, and do it like `{Binding Owner.RemoveCommand}`?

**Reply by MihaMarkic** [reply]
Good points.
1. Example is simplified to illustrate the issue. ChildView would be a separate view, also child could be hosted by different parents.
2. That's an option. However it would require new properties and more binding. Even more so when hierarchy is not direct (other children in between).

As per implementation, I don't think it's that difficult. Partially it is already there (through $parent) while the difference would be the condition - instead comparing view's type, it should compare it's DataContext's type.
